pkg/prometheus/promcfg.go

```diff
@@ -2520,6 +2520,8 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		})
 	}
 
+	cfg = append(cfg, yaml.MapItem{Key: "metric_relabel_configs", Value: generateRelabelConfig(labeler.GetRelabelingConfigs(sc.TypeMeta, sc.ObjectMeta, sc.Spec.MetricRelabelConfigs))})
+
```

Comment:
Most ScrapeConfig unit tests are failing as we're passing a new empty `metric_relabel_config` field to the configuration if it's not set. The test cases you updated added the key but not at the right spot. The following ensures that we're adding the key only if necessary.

```suggestion
      if sc.Spec.MetricRelabelConfigs != nil {
	cfg = append(cfg, yaml.MapItem{Key: "metric_relabel_configs", Value: generateRelabelConfig(labeler.GetRelabelingConfigs(sc.TypeMeta, sc.ObjectMeta, sc.Spec.MetricRelabelConfigs))})
      }
```

Reply:
Oops. My bad on missing the generated files. I've added this check and the optional annotation below and have now included the generated files